### ISSUE TITLE
can't import autosklearn.classification AutoSklearnClassifier

### ISSUE THREAD

**Issue by Motorrat** [automl/auto-sklearn#128]
I have re-installed anaconda and autosklearn because the upgrade failed due to insufficient space. Now I can not get AutoSklearnClassifier imported at all.
```
    from autosklearn.classification import AutoSklearnClassifier
  File "/home/user/anaconda2/lib/python2.7/site-packages/autosklearn/classification.py", line 1, in <module>
    from autosklearn.estimators import AutoSklearnClassifier
  File "/home/user/anaconda2/lib/python2.7/site-packages/autosklearn/estimators.py", line 8, in <module>
    import autosklearn.automl
  File "/home/user/anaconda2/lib/python2.7/site-packages/autosklearn/automl.py", line 27, in <module>
    from autosklearn.smbo import AutoMLSMBO
  File "/home/user/anaconda2/lib/python2.7/site-packages/autosklearn/smbo.py", line 14, in <module>
    from smac.smbo.smbo import SMBO, get_types
  File "/home/user/anaconda2/lib/python2.7/site-packages/smac/smbo/smbo.py", line 17, in <module>
    from smac.smbo.intensification import Intensifier
  File "/home/user/anaconda2/lib/python2.7/site-packages/smac/smbo/intensification.py", line 11, in <module>
    from smac.tae.execute_ta_run_aclib import ExecuteTARunAClib
  File "/home/user/anaconda2/lib/python2.7/site-packages/smac/tae/execute_ta_run_aclib.py", line 7, in <module>
    from smac.stats.stats import Stats
  File "/home/user/anaconda2/lib/python2.7/site-packages/smac/stats/stats.py", line 87
    def print_stats(self, debug_out:bool=False):
                                   ^
SyntaxError: invalid syntax
```

same error is shown here
https://travis-ci.org/automl/SMAC3/jobs/155928254

installed with these commands
pip install -r https://raw.githubusercontent.com/automl/auto-sklearn/master/requirements.txt
pip install git+https://github.com/automl/auto-sklearn.git#egg=autosklearn

**Comment by Motorrat** [comment]
I have done a complete clean re-install of the whole stack (OS, Anaconda2, scikit-learn and autosklearn) and get the same error.

**Comment by mlindauer** [comment]
Hi,

I'm not directly involved in the development of auto-sklearn, but I'm involved in SMAC (which is used in auto-sklearn). Since we stopped to support Python 2 in SMAC, auto-sklearn also fails to run with Python 2.
Please use Anaconda 4 (which is based on Python 3.5) to avoid this error.

Best,
Marius

**Comment by Motorrat** [comment]
Thanks @mlindauer, this is good to know and will probably save me a ton of time. 
BTW, the error above seems to be just a typo. I have changed the line to read 
`def print_stats(self, debug_out=False):` and it compiled.

**Comment by mlindauer** [comment]
no, this was not typo, but an annotation for type-checking.

**Comment by Motorrat** [comment]
This can be closed. I have re-installed with Anaconda3 and this error does not appear any more.

**Comment by mfeurer** [comment]
Sorry, I did not update the auto-sklearn master branch to reflect the SMAC changes. I'm working on a fix which makes the auto-sklearn also work with python2.7 for now. python2.7 support will be dropped in the next merge from development into master.

**Comment by mfeurer** [comment]
Should be fixed as of 12a457e05d8a54bc97eef0c7c9befee8b9dd08fd
